processor/src/main/kotlin/br/com/zup/beagle/processor/ContextObjectExtensionsFileBuilder.kt

```diff
@@ -0,0 +1,311 @@
+        val typeElement = elementUtils.getTypeElement(finalElementType.toString())
+        val isContextObject = typeElement?.getAnnotation(ContextObject::class.java) != null
+        val typeElementTypeName = typeElement.asType().asTypeName().javaToKotlinType()
+
+        if (isContextObject) {
+            fileBuilder.addFunction(buildListAccessFun(propertyName, typeElementTypeName, property.isNullable()))
+        }
+
+        fileBuilder.addFunction(buildChangeListElementFunFor(propertyName, typeElementTypeName))
+        fileBuilder.addFunction(buildChangeListElementFunFor(propertyName, typeElementTypeName.asBindType()))
+    }
+
+    private fun buildNormalizerFun(classFields: List<Element>): FunSpec {
+        val contextObjects = getContextObjectsFields(classFields)
+        val normalizingCode = buildNormalizeFunCodeWith(contextObjects)
+        val builder = FunSpec.builder("normalize")
+            .receiver(classTypeName)
+            .addCode(normalizingCode)
+            .returns(classTypeName)
+
+        if (!isGlobal) {
+            builder.addParameter("id", String::class)
+        }
+
+        return builder.build()
+    }
+
+    private fun buildNormalizeFunCodeWith(contextObjectsFields: List<Element>): String {
+        if (contextObjectsFields.isNotEmpty()) {
+            val str = contextObjectsFields.fold("") { acc, contextObject ->
+                val name = contextObject.simpleName.toString()
+                val propertyName = if (contextObject.isNullable()) "$name?" else name
+                val contextIdStatement = if (isGlobal) "global" else "\${id}"
+
+                resolveNormalizeImport(contextObject)
+
+                if (typeUtils.isIterable(contextObject.asType())) {
+                    "$acc,\n    $name = $propertyName.mapIndexed { index, contextObject ->\n" +
+                        "        contextObject.normalize(id = \"$contextIdStatement.$name[\$index]\")\n" +
+                        "    }"
+                } else {
+                    "$acc,\n    $name = $propertyName.normalize(id = \"$contextIdStatement.$name\")"
+                }
+            }
+
+            return if (isGlobal) {
+                "return this.copy(    ${str.drop(1)}\n)"
+            } else {
+                "return this.copy(\n    id = id$str\n)"
+            }
+        }
+
+        return if (isGlobal) {
+            "return this"
+        } else {
+            "return this.copy(id = id)"
+        }
+    }
+
+    private fun resolveNormalizeImport(contextObject: Element) {
+        if (getPackage(element) != getPackage(contextObject)) {
+            fileBuilder.addImport(
+                getPackage(contextObject),
+                "normalize"
+            )
+        }
+    }
+
+    private fun getPackage(element: Element): String {
+        val mutableList = typeUtils.getFinalElementType(element.asType()).asTypeName().toString().split(".") as MutableList
```

Comment:
<table data-meta="generated_by_backend">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle-backend-kotlin/blob/1767037c5777b4b44c2b78a548a8a8a8514b5d6b/processor/src/main/kotlin/br/com/zup/beagle/processor/ContextObjectExtensionsFileBuilder.kt#L162"></span>Line detected that is longer than the defined maximum line length in the code style.</td>
    </tr>
  </tbody>
</table>